app/lib/actions/pulp3/repository/commit_upload.rb

```diff
@@ -10,8 +10,31 @@ def invoke_external_task
           repo = ::Katello::Repository.find(input[:repository_id])
           repo_backend_service = repo.backend_service(smart_proxy)
           uploads_api = repo_backend_service.core_api.uploads_api
-          upload_commit = repo_backend_service.core_api.upload_commit_class.new(sha256: input[:sha256])
-          output[:pulp_tasks] = [uploads_api.commit(input[:upload_href], upload_commit)]
+          duplicate_sha_artifact_list = ::Katello::Pulp3::Api::Core.new(smart_proxy).artifacts_api.list("sha256": input[:sha256])
+          duplicate_sha_artifact_href = duplicate_sha_artifact_list&.results&.first&.pulp_href
```

Comment:
Is there ever a chance that this 'first' duplicate artifact is actually associated with a content unit in pulp?  Does it matter if its the first or the second?

Reply:
or are these only ever going to be artifacts that were uploaded as part of *this* upload?

Reply:
So there can only be one artifact with a sha..But we can have multiple content units with the same artifact at different paths..

Reply:
The duplicate_sha_artifact_list can only have 0 or 1 artifacts in the results since we are looking up with sha which is unique.